src/App.tsx

```diff
@@ -1,37 +1,55 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './component/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">{clockName}</strong>
-
-        {' time is '}
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId: number = 0;
```

Comment:
`0` is a valid value for `setInterval`, so if interval with id `0` already exists in the window, it can be stopped by a component if `window.clearInterval(this.timerId)` was, for example, in the `componentDidMount()` method
`window.clearInterval(undefined)` wont cause an error nor do anything
```suggestion
  timerId: number | undefined;
```

Reply:
Yes, thank you, but maybe timerId: number | undefined is incorrect because setInterval always returns a number, not undefined. Thanks for brining this to my attention, maybe better to use ReturnType<typeof setInterval> | null for accurate typing.